demos/dev_mode_key_provisioning/include/aws_dev_mode_key_provisioning.h

```diff
@@ -23,87 +23,195 @@
+ *                                   This should be a string values.  See iot_pkcs11_config.h.
+ * \param[out] pxPublicKeyHandle     The object handle of PKCS #11 private key created.
+ *
+ * \return CKR_OK upon successful key creation.  PKCS #11 error code on failure.
+ * Note that PKCS #11 error codes are positive.
+ */
+CK_RV xProvisionPublicKey( CK_SESSION_HANDLE xSession,
+                           uint8_t * pucKey,
+                           size_t xKeyLength,
+                           CK_KEY_TYPE xPublicKeyType,
+                           uint8_t * pucPublicKeyLabel,
+                           CK_OBJECT_HANDLE_PTR pxPublicKeyHandle );
+
 
 /** \brief Imports a certificate into the PKCS #11 module.
- * 
+ *
  * \param[in] xSession              A valid PKCS #11 session handle.
- * \param[in] pucCertificate        Pointer to a PEM certificate.  
- *                                  See tools/certificate_configuration/PEMfileToCString.html 
-                                    for help with formatting.
+ * \param[in] pucCertificate        Pointer to a PEM certificate.
+ *                                  See tools/certificate_configuration/PEMfileToCString.html
+ *                                  for help with formatting.
  * \param[in] xCertificateLength    Length of pucCertificate, in bytes.
  * \param[in] pucLabel              PKCS #11 label attribute value for certificate to be imported.
  *                                  This should be a string value. See iot_pkcs11.h.
- * \param[out] xObjectHandle        Updated to provide the PKCS #11 handle for the certificate
+ *                                  This should be a string value. See iot_pkcs11_config.h.
+ * \param[out] pxObjectHandle       Updated to provide the PKCS #11 handle for the certificate
  *                                  that was imported.
- *  
- * \return CKR_OK if certificate import succeeded. 
+ *
+ * \return CKR_OK if certificate import succeeded.
  * Otherwise, a positive PKCS #11 error code.
- */   
+ */
 CK_RV xProvisionCertificate( CK_SESSION_HANDLE xSession,
                              uint8_t * pucCertificate,
                              size_t xCertificateLength,
                              uint8_t * pucLabel,
-                             CK_OBJECT_HANDLE_PTR xObjectHandle );
+                             CK_OBJECT_HANDLE_PTR pxObjectHandle );
+
+/** \brief Generates an RSA key pair.
+ *
+ * \param[in] xSession              A valid PKCS #11 session handle.
+ * \param[in] pucPrivateKeyLabel    PKCS #11 label attribute value for private key to be created.
+ *                                  This should be a string value. See iot_pkcs11_config.h.
+ * \param[in] pucPublicKeyLabel     PKCS #11 label attribute value for public key to be created.
+ *                                  This should be a string value. See iot_pkcs11_config.h.
+ * \param[in] pxPrivateKeyHandle    Updated to provide the PKCS #11 handle for the private key
```

Comment:
I.e. which therefore must be freed by the caller?

Reply:
no which should have a valid CK_OBJECT_HANDLE created by the caller.  I will update the wording to match that of pkcs.